File: CS/PrintCrosshairSample/Form1.cs
using System;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraBars;
using DevExpress.XtraCharts;
using DevExpress.XtraCharts.Printing;

namespace PrintCrosshairSample {
    public partial class MainForm : DevExpress.XtraBars.Ribbon.RibbonForm {
        private MainViewModel VM { get; set; } = new MainViewModel();

        public MainForm() {
            InitializeComponent();
        }

        private void OnChartObjectHotTracked(object sender, HotTrackEventArgs e) {
            if (!(e.Object is Series))
                e.Cancel = true;
        }

        private void Form1_Load(object sender, EventArgs e) {
            InitChart();
        }

        private void InitChart() {
            var template = chartControl.SeriesTemplate;
            template.SeriesDataMember = "Company";
            template.SetDataMembers("Category", "Value");
            StackedBarSeriesView view = new StackedBarSeriesView();
            view.FillStyle.FillMode = FillMode.Solid;
            template.View = view;

            chartControl.DataSource = VM.TotalIncome;
            chartControl.OptionsPrint.SizeMode = PrintSizeMode.Zoom;

            chartControl.ObjectHotTracked += this.OnChartObjectHotTracked;
            chartControl.MouseUp += this.OnChartControlMouseUp;
        }


        private void OnChartControlMouseUp(object sender, MouseEventArgs e) {
            var diagram = (XYDiagram)chartControl.Diagram;
            diagram.ShowCrosshair(e.Location);
        }

        private void OnShowPrintPreviewItemClick(object sender, ItemClickEventArgs e) {
            chartControl.ShowPrintPreview();
        }

        private void OnPrintCrosshairItemClick(object sender, ItemClickEventArgs e) {
            chartControl.OptionsPrint.PrintCrosshair = bciPrintCrosshair.Checked;
        }
    }
}

File: VB/PrintCrosshairSample/Form1.vb
Imports System
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraBars
Imports DevExpress.XtraCharts
Imports DevExpress.XtraCharts.Printing

Namespace PrintCrosshairSample

    Public Partial Class MainForm
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Private Property VM As MainViewModel = New MainViewModel()

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub OnChartObjectHotTracked(ByVal sender As Object, ByVal e As HotTrackEventArgs)
            If Not(TypeOf e.Object Is Series) Then e.Cancel = True
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            InitChart()
        End Sub

        Private Sub InitChart()
            Dim template = chartControl.SeriesTemplate
            template.SeriesDataMember = "Company"
            template.SetDataMembers("Category", "Value")
            Dim view As StackedBarSeriesView = New StackedBarSeriesView()
            view.FillStyle.FillMode = FillMode.Solid
            template.View = view
            chartControl.DataSource = VM.TotalIncome
            chartControl.OptionsPrint.SizeMode = PrintSizeMode.Zoom
            Me.chartControl.ObjectHotTracked += AddressOf Me.OnChartObjectHotTracked
            Me.chartControl.MouseUp += AddressOf Me.OnChartControlMouseUp
        End Sub

        Private Sub OnChartControlMouseUp(ByVal sender As Object, ByVal e As MouseEventArgs)
            Dim diagram = CType(chartControl.Diagram, XYDiagram)
            diagram.ShowCrosshair(e.Location)
        End Sub

        Private Sub OnShowPrintPreviewItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs)
            chartControl.ShowPrintPreview()
        End Sub

        Private Sub OnPrintCrosshairItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs)
            chartControl.OptionsPrint.PrintCrosshair = bciPrintCrosshair.Checked
        End Sub
    End Class
End Namespace
